rfc/0003-policy-signing.md

```diff
@@ -0,0 +1,440 @@
+That could be an individual user or an organization.
+
+We plan to allow end users to define an "allow list" of GitHub users/organizations
+that can be trusted.
+
+## Verifying policies
+
+Policy verification settings are going to be defined using a dedicated configuration
+file. This file format is going to be shared between Policy Server and kwctl.
+
+The proposed format is the following one:
+
+```yaml
+public_keys:
+  threshold: 2
+  keys:
+    alice: |
+      -----BEGIN PUBLIC KEY-----
+      MBFKHFDGHKIJH0CAQYIKoZIzj0DAQcDQgAEX0HFTtCfTtPmkx5p1RbDE6HJSGAVD
+      BVDF6SKFSF87AASUspkQsN3FO4iyWodCy5j3o0CdIJD/KJHDJFHDFIu6sA==
+      -----END PUBLIC KEY-----
+    bob: |
+      -----BEGIN PUBLIC KEY-----
+      AUYSDHKJFGSHGSHCAQYIKoZIzj0DAQcDQgAEX0HFTtCfTtPmkx5p1RbtwDE1EVzu
+      wjQs1cCRKb5Pz/yUspkQsN3FO4iyWodCy5j3o0CdIJD/1gvq98pf4IG9tA==
+      -----END PUBLIC KEY-----
+issuers:
+  "https://token.actions.githubusercontent.com":
+  - https://github.com/flavio/
+  - https://github.com/kubewarden/
+  "https://github.com/login/oauth":
+  - john@acme.org
+annotations:
+  env: prod
+  foo: bar
+```
+
+The file is composed by three sections, which we will analyze in the next
+sections.
+
+
+### Public keys
+
+This is a list of the public keys of the authors that are trusted:
+
+```yaml
+public_keys:
+  threshold: 2
+  keys:
+    alice: |
+      -----BEGIN PUBLIC KEY-----
+      MBFKHFDGHKIJH0CAQYIKoZIzj0DAQcDQgAEX0HFTtCfTtPmkx5p1RbDE6HJSGAVD
+      BVDF6SKFSF87AASUspkQsN3FO4iyWodCy5j3o0CdIJD/KJHDJFHDFIu6sA==
+      -----END PUBLIC KEY-----
+    bob: |
+      -----BEGIN PUBLIC KEY-----
+      AUYSDHKJFGSHGSHCAQYIKoZIzj0DAQcDQgAEX0HFTtCfTtPmkx5p1RbtwDE1EVzu
+      wjQs1cCRKb5Pz/yUspkQsN3FO4iyWodCy5j3o0CdIJD/1gvq98pf4IG9tA==
+      -----END PUBLIC KEY-----
+    john:|
+      -----BEGIN PUBLIC KEY-----
+      MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAELKhD7F5OKy77Z582Y6h0u1J3GNA+
+      kvUsh4eKpd1lwkDAzfFDs7yXEExsEkPPuiQJBelDT68n7PDIWB/QEY7mrA==
+      -----END PUBLIC KEY-----
+```
+
+* `keys`: list of keys that are trusted. The owner's name is not relevant during
+  the verification process.
+* `threshold`: an integer number of keys whose signatures are required in order
+  to consider a policy as being properly signed.
+
+Given the above example, there are three keys defined and the threshold is set to
+`2`. That means all the policies loaded must be signed by two or more of these
+keys.
+
+This gives some resiliency if one of these keys is compromised.
+
+### Issuers
+
+It allows to define, on a per issuer basis, who are the trusted`Subject`s:
+
+```yaml
+issuers:
+  "https://token.actions.githubusercontent.com":
+  - https://github.com/flavio/
+  - https://github.com/kubewarden/
+  - https://github.com/alice/fancy-policy/
+  "https://github.com/login/oauth":
+  - john@acme.org
+```
+
+The string entered by the user is then used to match the beginning of the `Subject`
+found inside of a signature.
```

Comment:
I think it should be interpreted as an URL, and matched as an URL when checked. So, we always automatically append the trailing slash. URL's are always checked with trailing slashed appended. Comparison is not a string-based one.

This helps with your concern in https://github.com/kubewarden/kubewarden-controller/pull/147/files#r790110976